ratelimits/limiter.go

```diff
@@ -83,6 +87,96 @@ type Decision struct {
 	// theoretical arrival time (TAT) of next request. It must be no more than
 	// (burst * (period / count)) in the future at any single point in time.
 	newTAT time.Time
+
+	// transaction is the Transaction that resulted in this Decision. It is
+	// included for the production of verbose Subscriber-facing errors. It is
+	// set by the Limiter before returning the Decision.
+	transaction Transaction
+}
+
+// RateLimitError returns a verbose Subscriber-facing error for denied
+// *Decisions. If the *Decision is allowed, it returns nil.
+func (d *Decision) RateLimitError(clk clock.Clock) error {
```

Comment:
I was considering making almost all of the fields in the Decision private and then renaming `RateLimitError()` something like `Allowed()` which returns `(bool, error)` and instructing the `Limiter` caller to call `Allowed()` on the resulting decision. What do you think?

Reply:
Ooh hmm. I almost really like it.

It's true that the Remaining, RetryIn, and ResetIn fields aren't used anywhere outside the ratelimits package. So going the rest of the way and making this type's fields wholly private makes sense to me.

I think the only bit the feels slightly off is the `(bool, error)` return type. The data in those is redundant -- if the bool is false, the error will always be non-nil, and vice versa. But having a method named `.Allowed()` just return an error feels a bit off, and having to make two calls (one to .Allowed(), and then a follow-up to get the error if allowed is false) also feels inelegant.

So overall totally in favor, still mulling over what the best API is.

Reply:
> still mulling over what the best API is.

Please let me know if something strikes you as feeling right. What I have now works but feels obtuse. FWIW, I don't want perfect to get in the way of done either.

Reply:
I think what you have here and now is definitely good enough to land. I'd just consider making the fields which aren't used outside this package private. It'll be an easy refactor if we come up with a better API later.